python/ray/tests/conftest.py

```diff
@@ -201,6 +204,31 @@ def redis_alive(port, enable_tls):
     return False
 
 
+def start_redis_with_sentinel(db_dir):
+    temp_dir = ray._private.utils.get_ray_temp_dir()
+
+    redis_ports = find_available_port(49159, 55535, redis_sentinel_replicas() + 1)
+    sentinel_port = redis_ports[0]
+    master_port = redis_ports[1]
+    redis_processes = [
+        start_redis_instance(temp_dir, p, listen_to_localhost_only=True, db_dir=db_dir)[
+            1
+        ]
+        for p in redis_ports[1:]
+    ]
+    # setup replicas of the master
+
+    time.sleep(1)
```

Comment:
Can we do better than a sleep: i.e. more deterministic

Reply:
Good point! I updated to a wait function to check all redis instances are up.